### DISCUSSION TITLE
Returning dates (and other custom deserialization) in invoke response

### DISCUSSION THREAD

**Discussion by StelFoog** [tauri-apps/tauri#7196] [Q&A] [answered]
Is there a way to make the response from invoke calls have custom deserialization? I'm primarily looking to convert strings in _ISO 8601_ like `"2023-06-14T09:22:52.348Z"` into `Date` objects

**Comment by Kushagra9671** [comment]
Good

**Comment by Kushagra9671** [comment]
Very good

**Comment by FabianLars** [comment]
> Is there a way to make the response from invoke calls have custom deserialization?

Yes. The commands only care about the types implementing `serde::Deserialize`, what that deserialization actually looks like doesn't really matter.
In case your Date object doesn't implement Deserialize, or not in a way that works for you, you'll have to implement it yourself https://serde.rs/impl-deserialize.html.

**Reply by FabianLars** [reply, marked as answer]
Sorry i mixed things up. You meant deserialization on the javascript side right? So that will be a bit more tricky since Tauri only calls JSON.parse and calls it a day. In this case you'll likely have to wrap around `invoke` and do it yourself since you can't add a reviver to the parse call Tauri uses internally.
